rust/connlib/tunnel/src/lib.rs

```diff
@@ -199,13 +175,7 @@ pub struct Tunnel<C: ControlSignal, CB: Callbacks, TRoleState> {
     public_key: PublicKey,
     peers_by_ip: RwLock<IpNetworkTable<Arc<Peer>>>,
     peer_connections: Mutex<HashMap<ConnId, Arc<RTCPeerConnection>>>,
-    awaiting_connection: Mutex<HashMap<ConnId, AwaitingConnectionDetails>>,
-    gateway_awaiting_connection: Mutex<HashMap<GatewayId, Vec<IpNetwork>>>,
-    resources_gateways: Mutex<HashMap<ResourceId, GatewayId>>,
     webrtc_api: API,
-    resources: Arc<RwLock<ResourceTable<ResourceDescription>>>,
-    control_signaler: C,
-    gateway_public_keys: Mutex<HashMap<GatewayId, PublicKey>>,
```

Comment:
@conectado I think the path forward is to not have any shared data between the client and gateway. Moving `peers_by_ip` and `peer_connections` into `ClientState` and `GatewayState` will remove the last two locks. We can always have shared _functionality_ between the two by having functions that take the unlocked datastructure by reference so there won't be much duplication apart from a field declaration in each state variable.

Once each end has full control over all the data in one place, I think we can encapsulate it in a nicer way and make all the fields private.

Thoughts?

Reply:
I'll do this as a follow-up straight after this PR. I think it will actually turn out super nicely :)